.github/workflows/ci.yml

```diff
@@ -105,32 +114,51 @@ jobs:
           - 6379:6379
         options: --entrypoint redis-server
     steps:
-      - uses: actions/checkout@v3
+      - uses: actions/checkout@v4
       - name: Set up Node.js
-        uses: actions/setup-node@v3
+        uses: actions/setup-node@v4
         with:
-          node-version: 18.15.0
+          node-version-file: .node-version
       - name: Set up Ruby
         uses: ruby/setup-ruby@v1
         with:
-          ruby-version: 3.3.3
+          ruby-version: .ruby-version
           bundler-cache: true
+      - name: Cache js dependencies
+        uses: actions/cache@v4
+        with:
+          key: node_modules-${{ hashFiles('yarn.lock') }}
+          path: node_modules
       - name: Install JS dependencies
         run: yarn install
       - name: Precompile assets
         run: RAILS_ENV=test bundle exec rake assets:precompile
-      - name: Setup test database
-        env:
-          POSTGRES_USER: rdv_insertion_test
-          POSTGRES_PASSWORD: rdv_insertion_test
-          POSTGRES_DB: rdv_insertion_test
-        run: bundle exec rails db:create db:schema:load RAILS_ENV=test
-      - name: Run specs
+      - name: List all feature spec files
+        run: |
+          mkdir -p tmp
+          find spec/features -name "*_spec.rb" > tmp/feature_spec_files.txt
+          echo "Total feature spec files found:"
+          wc -l tmp/feature_spec_files.txt
+      - name: Setup test database and run specs
+        run: |
+          echo "Spec files for this node:"
+          cat tmp/feature_spec_files.txt | sort | awk "NR % $CI_NODE_TOTAL == $CI_NODE_INDEX" | tee node_spec_files.txt
```

Comment:
Les tests sont divisés entre les node de la matrice, on équilibre vers chaque node les tests en fonction de leur numero de ligne dans le fichier `feature_spec_files.txt`

Reply:
Pour documenter: cette répartition est faite séquentiellement (ligne 1 -> node 0, ligne 2 -> node 1, etc.. ), d'où la répartition inégale du temps d'exécution de chaque node